### DISCUSSION TITLE
What does import with tilde (~) means?

### DISCUSSION THREAD

**Discussion by robertwt7** [remix-run/remix#733] [Q&A] [answered]
I found [this answer](https://stackoverflow.com/questions/43113870/what-is-the-tilde-doing-in-this-javascript-import) from stackoverflow but it doesn't answer my question.

I found a lot of code with tilde import in the Remix tutorial, i.e: `import type { Post } from "~/posts"`

Just wondering if anyone knows what does it mean? Which path does the tilde (~) resolves to in this case?

**Comment by mcansh** [comment, marked as answer]
the tilde is an alias for your `./app` directory, in this case `~/posts` maps to `./posts.{ts,tsx,js,jsx,cts,mts}` or `./posts/index.{ts,tsx,js,jsx,cts,mts}` in your app directory https://github.com/remix-run/remix/blob/main/packages/create-remix/templates/_shared_ts/tsconfig.json#L11-L13. We currently only support this path alias, but are [looking](https://github.com/remix-run/remix/pull/320) into expanding it to any alias you configure in your jsconfig or tsconfig

**Reply by robertwt7** [reply]
ah right. yeah i figure after looking at the `tsconfig.json` file.  Thanks for the answer

This may be off topic, but do you know how to adjust vscode settings to prevent it from complaining?

![image](https://user-images.githubusercontent.com/15647967/143733738-17c822e0-ac58-44d6-86b6-6b2ac16ef104.png)

**Reply by silvenon** [reply]
You're getting a TypeScript error, which is unexpected, I assume you'd be getting the same error if you ran `npx tsc`, in that case it's not a VS Code problem. Are you sure that this file exists? `app/posts/post.tsx` or something similar.

**Reply by robertwt7** [reply]
@silvenon yeah it does. I continued the quickstart by changing to `import { getPost } from "./posts/post"`
